yb-voyager/cmd/exportData.go

```diff
@@ -1780,12 +1781,29 @@ func createUpdateExportedRowCountEventList(tableNames []string) []*cp.UpdateExpo
 	return result
 }
 
-func saveTableToUniqueKeyColumnsMapInMetaDB(tableList []sqlname.NameTuple) {
+func saveTableToUniqueKeyColumnsMapInMetaDB(tableList []sqlname.NameTuple, leafPartitions *utils.StructMap[sqlname.NameTuple, []sqlname.NameTuple]) {
 	res, err := source.DB().GetTableToUniqueKeyColumnsMap(tableList)
 	if err != nil {
 		utils.ErrExit("get table to unique key columns map: %w", err)
 	}
 
+	//Adding all the leaf partitions unique key columns to the root table unique key columns since in the importer all the events only have the root table name
+	leafPartitions.IterKV(func(key sqlname.NameTuple, value []sqlname.NameTuple) (bool, error) {
+		keyTbl := key.AsQualifiedCatalogName()
```

Comment:
this is a bit error-prone. 
`GetTableToUniqueKeyColumnsMap` is returning `map[string][]string` so the key structure is not fixed. IF we change the implementation of `GetTableToUniqueKeyColumnsMap` in the future, this will break. 

Ideally, it should be the case that: 
1. `GetTableToUniqueKeyColumnsMap` returns `StructMap[NameTuple][]string`
2. then i can directly just operate on a common key.